### PR TITLE
Use uv in contribution document

### PR DESCRIPTION
**Pull request by zanieb**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -29,16 +29,14 @@ You'll also need [Insta](https://insta.rs/docs/) to update snapshot tests:
...
-```
+You'll need [uv](https://docs.astral.sh/uv/getting-started/installation/) (or `pipx` and `pip`) to
+run Python utility commands.
 
 You can optionally install pre-commit hooks to automatically run the validation checks
 when making a commit:
 
 ```shell
+uv tool install pre-commit
 pre-commit install
```
Comment (MithicSpirit) [inline CONTRIBUTING.md:40]: Shouldn't this be `uvx pre-commit install`, as it won't necessarily be the first result for pre-commit on the PATH?

Comment (zanieb) [reply CONTRIBUTING.md:40]: If you `uvx pre-commit install` won't the pre-commit executable be missing during the git hook invocations?

Comment (MithicSpirit) [reply CONTRIBUTING.md:40]: Just tested it in a clean docker container, and `uvx pre-commit install` works. It uses the python that it was originally executed with, so it doesn't need to be in the PATH (although it falls back to that if the original python is no longer available for some reason).

Comment (zanieb) [reply CONTRIBUTING.md:40]: I assume this relies on an environment that's in the uv cache though — that's generally bad practice.

**Pull request merged**